### PR TITLE
Getting CommonCrypto and playgrounds to play nicely

### PR DESCRIPTION
**Pull request by mbrandonw**
### What

Right now playgrounds are busted cause of commoncrypto. Why is commoncrypto even in this repo? Good question! It probably shouldn't! But we'll move it out another day.

In the meantime, if you copy a module map to a particular location on your computer, it will all work.

Even tho commoncrpyto should probably be removed from this repo, this work will still be pertinent cause we need to do something similar in pointfreeco.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -13,6 +13,13 @@ imports = \
 xcodeproj:
 	swift package generate-xcodeproj
 
+bootstrap: common-crypto-mm xcodeproj
+
+common-crypto-mm:
+	-@mkdir "$(COMMON_CRYPTO_PATH)" || true
+	-@touch "$(COMMON_CRYPTO_MODULE_PATH)" || true
+	-@echo "$$COMMON_CRYPTO_MODULE_MAP" > "$(COMMON_CRYPTO_MODULE_PATH)"
```
Comment (mbrandonw) [inline Makefile]: so this creates the `CommonCrypto.framework` directory, then adds a module map file, and then puts the module map code into the file.

File: Makefile [hunk 2]
```diff
@@ -43,3 +50,16 @@ test-swift:
 	swift test
 
 test-all: test-linux test-mac test-ios
+
+SDK_PATH = $(shell xcrun --show-sdk-path)
```
Comment (mbrandonw) [inline Makefile]: the place we install the module map is dependent on what Xcode you have selected (via `xcode-select`)

File: Web.playground/Pages/HttpPipeline.xcplaygroundpage/Contents.swift [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import Foundation
+import Html
+import HttpPipeline
```
Comment (mbrandonw) [inline Web.playground/Pages/HttpPipeline.xcplaygroundpage/Contents.swift]: `HttpPipeline` is the framework with the commoncrypto dependency, and so importing this used to break playgrounds. now it all works!

File: Makefile [hunk 4]
```diff
@@ -13,6 +13,13 @@ imports = \
 xcodeproj:
 	swift package generate-xcodeproj
 
+bootstrap: common-crypto-mm xcodeproj
+
+common-crypto-mm:
+	-@mkdir "$(COMMON_CRYPTO_PATH)" || true
```
Comment (stephencelis) [inline Makefile]: `mkdir -p` should negate the need for `|| true`

File: Makefile [hunk 5]
```diff
@@ -13,6 +13,13 @@ imports = \
...
+	-@touch "$(COMMON_CRYPTO_MODULE_PATH)" || true
```
Comment (stephencelis) [inline Makefile]: This should never fail, no? And if you're `echo`ing the next line, maybe it's not even necessary?

File: Makefile [hunk 6]
```diff
@@ -43,3 +50,16 @@ test-swift:
 	swift test
 
 test-all: test-linux test-mac test-ios
+
+SDK_PATH = $(shell xcrun --show-sdk-path)
+FRAMEWORKS_PATH = $(SDK_PATH)/System/Library/Frameworks
+COMMON_CRYPTO_PATH = $(FRAMEWORKS_PATH)/CommonCrypto.framework
+COMMON_CRYPTO_MODULE_PATH = $(COMMON_CRYPTO_PATH)/module.map
+define COMMON_CRYPTO_MODULE_MAP
```
Comment (stephencelis) [inline Makefile]: Whohohoa! Makefile expert! Never seen this before!

File: Makefile (hunk 4 above)
Comment (mbrandonw) [reply Makefile]: lol very true

File: Makefile (hunk 5 above)
Comment (mbrandonw) [reply Makefile]: ha... yahhhhh

File: Makefile (hunk 6 above)
Comment (mbrandonw) [reply Makefile]: 💅

**Pull request merged**